Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +423,245 @@ export class CppSettings extends Settings {
+    // Returns the value of a setting as a boolean with proper type validation.
+    private getAsBoolean(settingName: string): boolean;
+    private getAsBoolean(settingName: string, allowNull: boolean): boolean | null;
+    private getAsBoolean(settingName: string, allowNull: boolean = false): boolean | null {
+        const value: any = super.Section.get<boolean | null>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        if (isBoolean(value)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
     }
 
-    public get vcFormatSpaceAroundBinaryOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundBinaryOperator")!;
+    // Returns the value of a setting as a string with proper type validation and checks for valid enum values.
+    private getAsString(settingName: string): string;
+    private getAsString(settingName: string, allowNull: boolean): string | null;
+    private getAsString(settingName: string, allowNull: boolean = false): string | null {
+        const value: any = super.Section.get<string | null>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (setting.enum !== undefined) {
+            if (this.isValidEnum(setting.enum, value)) {
+                return value;
+            }
+            return setting.default;
+        }
+        if (isString(value)) {
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpaceAroundAssignmentOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundAssignmentOperator")!;
+    // Returns the value of a setting as a number with proper type validation and checks if value falls within the specified range.
+    private getAsNumber(settingName: string): number;
+    private getAsNumber(settingName: string, allowNull: boolean): number | null;
+    private getAsNumber(settingName: string, allowNull: boolean = false): number | null {
+        const value: any = super.Section.get<number | null>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (value === null) {
+            return setting.default;
+        }
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
```

Comment:
This check is not needed. If it's null, `isNumber` will return false and the default setting value will be returned.

Reply:
This comment landed in the wrong place. I was talking about the `value === null` check 3 lines up.